cni/pkg/install-cni/cmd/root.go

```diff
@@ -0,0 +1,131 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"istio.io/istio/cni/pkg/install-cni/pkg/config"
+	"istio.io/istio/cni/pkg/install-cni/pkg/constants"
+	"istio.io/istio/cni/pkg/install-cni/pkg/install"
+	"istio.io/pkg/log"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "install-cni",
+	Short: "Install and configure CNI on a node",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cfg, err := constructConfig()
+		if err != nil {
+			return err
+		}
+		return install.Run(cfg)
+	},
+}
+
+// GetCommand returns the main cobra.Command object for this application
+func GetCommand() *cobra.Command {
+	return rootCmd
+}
+
+func init() {
+	viper.AutomaticEnv()
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+
+	registerStringParameter(constants.CNINetDir, "/etc/cni/net.d", "Directory on the host where CNI networks are installed")
+	registerStringParameter(constants.CNIConfName, "", "Name of the CNI configuration file")
+	registerBooleanParameter(constants.ChainedCNIPlugin, true, "Whether to install CNI plugin as a chained or standalone")
+	registerStringParameter(constants.CNINetworkConfig, "", "CNI config template as a string")
+
+	// Not configurable in CNI helm charts
+	registerStringParameter(constants.MountedCNINetDir, "/host/etc/cni/net.d", "Directory on the container where CNI networks are installed")
+	registerBooleanParameter(constants.Sleep, true,
```

Comment:
Should this configuration exist at all? It should always watch the files and never return, no?

Reply:
I'm not sure if it is ever used (maybe used for testing?), but that configuration existed before I refactored the shell script in [another PR](https://github.com/istio/istio/blob/7d6670ed88b872929c3df8fb50c81e123fef7f72/cni/deployments/kubernetes/install/scripts/install-cni.sh#L392). If you check the diff, I kept that same configuration that existed before.
Should I just remove this configuration?

Reply:
+1 from removing it. It sounds like a hack, or an implementation detail. It should not be exposed to users.

Reply:
Should I just remove the viper option or remove all of the logic associated with that config? If I just remove the viper option I can rename that field to something that makes more sense in the Config struct.